### PR TITLE
feat: Add EIP-6963 Provider

### PR DESCRIPTION
**Pull request by jiexi**
# **Description**

Bump `@metamask/providers` to v13.1.0 which includes EIP6963 support. Pass necessary provider info values to the inpage bridge script to properly initialize the provider.

## **Manual testing steps**

* Visit eip6963.org
* MetaMask Mobile should show up in the list
* Click Connect
* Should show currently connected network

## **Screenshots/Recordings**

Screenshot 2023-10-18 at 4 04 27 PM

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/869

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Cal-L — DISMISSED** [review]
LGTM

File: scripts/inpage-bridge/inpage/webpack.config.js [hunk 1]
```diff
@@ -39,6 +47,11 @@ const config = {
       Buffer: ['buffer', 'Buffer'],
       process: 'process/browser',
     }),
+    new webpack.DefinePlugin({
+      'process.env.METAMASK_BUILD_NAME': JSON.stringify('MetaMask Mobile'),
+      'process.env.METAMASK_BUILD_ICON': JSON.stringify(getBuildIcon()),
+      'process.env.METAMASK_BUILD_APP_ID': JSON.stringify('io.metamask.mobile'),
```
Comment (jiexi) [inline scripts/inpage-bridge/inpage/webpack.config.js:53]: Is the name and rdns value what we want here?

Comment (jiexi) [reply scripts/inpage-bridge/inpage/webpack.config.js:53]: @rekmarks

File: scripts/inpage-bridge/inpage/webpack.config.js [hunk 2]
```diff
@@ -1,5 +1,13 @@
 const webpack = require('webpack');
 const path = require('path');
+const { readFileSync } = require('fs');
+
+const SVG_LOGO_PATH =
+  '../../../app/component-library/components/Icons/Icon/assets/fox.svg';
```
Comment (jiexi) [inline scripts/inpage-bridge/inpage/webpack.config.js:6]: is this logo okay to use?

**Comment by jiexi** [comment]
The assumption here is that Mobile Flask is built off a different branch, thus no build time variables are needed to determine the EIP6963 provider name, icon, and rdns values

File: scripts/inpage-bridge/inpage/webpack.config.js (hunk 2 above)
Comment (jiexi) [reply scripts/inpage-bridge/inpage/webpack.config.js:6]: I don't see why not. shows up fine, and build will fail if it is moved.

File: scripts/inpage-bridge/inpage/webpack.config.js (hunk 1 above)
Comment (jiexi) [reply scripts/inpage-bridge/inpage/webpack.config.js:53]: to make sure we make the mobile RC cut tomorrow, I'm going to go with
name: MetaMask
rdns: io.metamask.mobile

My reasoning is that the name just needs to be a user friendly identifier, additionally this aligns us with the app name which is just `MetaMask`. RDNS value should be unique to help dapps/sdk identify exactly what metamask provider they are interacting with

**Review by Cal-L — APPROVED** [review]
LGTM

**Comment by jiexi** [comment]
Pulled down `release/7.10.0` and tested locally

https://github.com/MetaMask/metamask-mobile/assets/918701/46d2b8e7-e7e9-4b88-924a-61816f2ca9f4

**Pull request merged**